odoo/addons/base/models/res_users.py

```diff
@@ -510,8 +511,8 @@ def read(self, fields=None, load='_classic_read'):
 
     def _has_field_access(self, field, operation):
         return super()._has_field_access(field, operation) or (
-            self == self.env.user
-            and operation == 'read'
+            operation == 'read'
+            and (self == self.env.user or self._origin == self.env.user)
```

Comment:
When `self` is real records, `self._origin == self`.
```suggestion
            and self._origin == self.env.user
```
If it's a performance issue, then I suggest to optimize `records._origin` instead.  Simply return `records` when all its `_ids` are truthy.  There are several tests on `record._origin` in `Fields.__get__` that may benefit from that optimization, don't you think?

Reply:
I don't this it is a performance issue here. I will make a separate PR for that.
(I adapted the code here)